Original author: Julian Ospald <hasufell@gentoo.org>
Reworked for v0.6 by: Manuel Rueger <gentoo@rueg.eu>

Set CC
Remove uncommon CFLAGS
Make NLS optional
Drop gitver.sh
Reorder flags
Strip binaries

--- Makefile.old	2012-08-30 02:04:32.672389762 +0200
+++ Makefile	2012-08-30 02:33:37.390251104 +0200
@@ -9,20 +9,21 @@
 obj = $(src:.c=.o)
 verh = include/version.h
 
+CC ?= gcc
+
 CFLAGS ?= -Os
 CFLAGS += -Wall -Wextra -pthread
 CFLAGS += -Iinclude
-CFLAGS += -ffunction-sections -fdata-sections
 CFLAGS += $(shell pkg-config --cflags pciaccess)
 CFLAGS += $(shell pkg-config --cflags ncurses 2>/dev/null)
 
 # Comment this if you don't want translations
-CFLAGS += -DENABLE_NLS=1
+ifdef nls
+	CPPFLAGS += -DENABLE_NLS=1
+endif
 
 ifdef debug
 	CFLAGS += -g
-else
-	CFLAGS += -s
 endif
 
 LDFLAGS += -Wl,-O1 -Wl,-gc-sections
@@ -38,7 +39,7 @@
 $(obj): $(wildcard include/*.h) $(verh)
 
 $(bin): $(obj)
-	$(CC) -o $(bin) $(obj) $(CFLAGS) $(LDFLAGS) $(LIBS)
+	$(CC) $(CFLAGS) $(CPPFLAGS) -o $(bin) $(obj) $(LDFLAGS) $(LIBS)
 
 clean:
 	rm -f *.o $(bin)
@@ -46,7 +47,7 @@
 .git:
 
 $(verh): .git
-	./getver.sh
+#	./getver.sh
 
 trans:
 	xgettext -o translations/radeontop.pot -k_ *.c \
@@ -57,6 +58,9 @@
 	$(INSTALL) -D -m644 radeontop.1 $(DESTDIR)/$(PREFIX)/share/man/man1/radeontop.1
 	$(MAKE) -C translations install PREFIX=$(PREFIX)
 
+strip:
+	strip --strip-unneeded -R .comment -R .GCC.command.line $(bin)
+
 man:
 	a2x -f manpage radeontop.asc
 
